bocoup/threeflows
# Researcher: fix data download (contents), match "My Scenario Data" UI

https://app.asana.com/0/1127815256483386/1154060543570009/f

In addition to that issue, I realized that the data available here wasn't complete. It was also painful to use the dropdown, so I just made it a table
------------------------------------------------------------------------

This data was supposed to be "gatherable" by specifying a specific cohort of runs to download—that capability is not implemented yet as I ran out of time today and will implement as a follow up either tonight or tomorrow morning. 



------------------------------------------------------------------------

Pre-test: 

- Log in as a user with Researcher role/permissions
- Click "Research" in the top navigation menu

Test 1: 

- Click the numbered pagination buttons
- Observe that you can navigate through many scenarios

Test 2: 

- Click the "Download" icon\* of any scenario to download the entire consented history of responses for a given scenario


Test 3: 

- Click on one of the scenario titles, say "No" to consent, then complete a run.
- Return to the Research view, click the "Download" icon\*, view the data and confirm that your responses are not present. 

Test 4: 

- Click on one of the scenario titles, edit the url in the address bar to add "?a=1&b=2", then hit enter to load the scenario run. Click "Yes" to consent, then complete a run.
- Return to the Research view, click the "Download" icon\*, view the data and confirm that your responses are present and include the referrer params that you added to the url.



------------------------------------------------------
\* "Download" icon: 
![image](https://user-images.githubusercontent.com/27985/71013997-36e45f00-20bf-11ea-9491-928d7a63e602.png)
